Application/EdFi.Ods.Api/Container/Modules/ApplicationModule.cs

```diff
@@ -195,6 +196,10 @@ protected override void Load(ContainerBuilder builder)
                 .EnableInterfaceInterceptors()
                 .SingleInstance();
 
+            builder.RegisterType<ApiClientDetailsCacheKeyProvider>()
```

Comment:
This class is now only used by the ExternalCache feature:
![image](https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS/assets/1467340/c0e2fe29-35fd-47e7-8c9f-06241934dd78)

So let's move the interface and implementation classes (`IApiClientDetailsCacheKeyProvider` and `ApiClientDetailsCacheKeyProvider`) to the ExternalCache folder in the Features project, and also move this registration should move to the `ExternalCacheModule` class.